### DISCUSSION TITLE
Issue with writing Avro files directly to S3 bucket

### DISCUSSION THREAD

**Discussion by clarence-coppinger** [aws/aws-sdk-java#3049] [General]
When using apache Avro library to write Avro files directly to s3 bucket using the AmazonS3#[putObject](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/s3/AmazonS3.html#putObject-java.lang.String-java.lang.String-java.io.InputStream-com.amazonaws.services.s3.model.ObjectMetadata-)([String](http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true) bucketName, [String](http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true) key, [InputStream](http://docs.oracle.com/javase/7/docs/api/java/io/InputStream.html?is-external=true) input, [ObjectMetadata](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/s3/model/ObjectMetadata.html) metadata) method. 

Upon trying to read the files from the bucket using Avro tools (command line tool to read Avro files) we see this exception:  “org.apache.avro.AvroRuntimeException: Malformed data. Length is negative: -25669624”

When we use a similar process to write files to GCS the files are able to be processed after being uploaded without issue. My question here is: Are we using the wrong method to upload these files? When we write the Avro files locally and upload them we can read them afterwards but trying to write them directly to the bucket seems to cause an issue.

**Comment by debora-ito** [comment]
Hey @clarence-coppinger we recommend you reach out to the Avro team with this issue. We can't help troubleshooting issues with third-party libraries.

**Reply by clarence-coppinger** [reply]
Just to clarify this isn't an issue with the avro library. I referenced the putObject method from the AWS SDK and was having an issue with avro files that were uploaded through the AWS S3 API.
